### PR TITLE
ltc: ctr: update pt and ct after acceleration

### PR DESCRIPTION
**Pull request by vchong**
Problem occurs in the condition of the following case:

1st decryption:
Decrypt a ciphertext whose length is a multiple of the block size (16B)
(len = n * block_size)
2nd decryption:
Decrypt the continuing ciphertext whose length is not a multiple of the
block size
(len = m * block_size + l)

In this case accel_ctr_encrypt() is firstly used at the 2nd decryption.
If pt and ct are not updated, the top (l = len % block_size) bytes of
decryption result are sometimes destroyed.

Reported-by: Tetsuya Yoshizaki <yoshizaki.tetsuya@socionext.com>
Signed-off-by: Tetsuya Yoshizaki <yoshizaki.tetsuya@socionext.com>
Signed-off-by: Victor Chong <victor.chong@linaro.org>

### CODE REVIEW

**Comment by vchong** [comment]
Is the `Reported-by` needed in this case?

**Comment by jforissier** [comment]
**TL;DR:** please just replace `Reported-by: ...` by `From: ...`

I suggest using a `From:` tag as the first entry in the tag section. Keep the original author's `Signed-off-by:`. Then I will merge the patch manually, removing the `From:` and passing the `--author` option to `git`.

Note, the Linux kernel has the same convention when it comes to sending patches to the mailing lists, except that the `From:` line should be the first in the message (see [here](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/submitting-patches.rst?h=v4.15-rc8#n688)). The `From:` does not appear in the final change log, instead it is picked up automatically by `git am`. This would probably not work for us because checkpatch considers the first line of the commit log to be the subject, so we might have issues with CI. That's why I propose to move the `From:` to the bottom.

**Comment by jforissier** [comment]
@sjaeckel FYI.

**Comment by vchong** [comment]
@jforissier Updated. Thanks!

**Comment by jenswi-linaro** [comment]
Here's a test case https://github.com/OP-TEE/optee_test/pull/248

**Comment by sjaeckel** [comment]
thx! c.f. https://github.com/libtom/libtomcrypt/commit/d1d3ae2d1e705f36e7d313aa4a9b61c0a146ee44

**Comment by jforissier** [comment]
@jenswi-linaro you beat me to it ;-)
It seems to me that the commit description in this PR is somewhat misleading. I understand you need to call the encrypt function only two times where in fact the error occurs on the third time if I'm not mistaken.

**Comment by jforissier** [comment]
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey960)`

**Comment by jforissier** [comment]
Merged manually. Thanks guys.

@jenswi-linaro I have converted your "LGTM" (that you gave via email) into an Acked-by:
I have also added a link to the upstream commit.
